x-pack/legacy/plugins/epm/server/lib/elasticsearch/template/install.ts

```diff
@@ -22,41 +23,53 @@ const isFields = (path: string) => {
  * in one datasets, they are merged together into 1 and then converted to a template
  * The template is currently loaded with the pkgey-package-dataset
  */
-export async function installTemplates(pkg: RegistryPackage, callCluster: CallESAsCurrentUser) {
-  // If no datasets exist in this package, no templates have to be installed.
-  if (!pkg.datasets) return;
-  return pkg.datasets.map(async dataset => {
-    dataset.package = pkg.name;
-    // Fetch all field definition files for this dataset
-    const fieldDefinitionFiles = await getAssetsData(pkg, isFields, dataset.name);
-    // Merge all the fields of a dataset together and create an Elasticsearch index template
-    let fields: Field[] = [];
-    for (const file of fieldDefinitionFiles) {
-      // Make sure it is defined as it is optional. Should never happen.
-      if (file.buffer) {
-        const tmpFields = safeLoad(file.buffer.toString());
-        // safeLoad() returns undefined for empty files, we don't want that
-        if (tmpFields) {
-          fields = fields.concat(tmpFields);
-        }
+export async function installTemplateForDataset(
+  pkg: RegistryPackage,
+  callCluster: CallESAsCurrentUser,
+  dataset: Dataset,
+  datasourceName: string
+) {
+  // Fetch all field definition files for this dataset
+  const fieldDefinitionFiles = await getAssetsData(pkg, isFields, dataset.name);
+  // Merge all the fields of a dataset together and create an Elasticsearch index template
+  let fields: Field[] = [];
+  for (const file of fieldDefinitionFiles) {
+    // Make sure it is defined as it is optional. Should never happen.
+    if (file.buffer) {
+      const tmpFields = safeLoad(file.buffer.toString());
+      // safeLoad() returns undefined for empty files, we don't want that
+      if (tmpFields) {
+        fields = fields.concat(tmpFields);
       }
     }
-    return installTemplate({ callCluster, fields, dataset });
-  });
+    dataset.package = pkg.name;
+    return installTemplate({ callCluster, fields, dataset, datasourceName });
+  }
 }
 
 async function installTemplate({
   callCluster,
   fields,
   dataset,
+  datasourceName,
 }: {
   callCluster: CallESAsCurrentUser;
   fields: Field[];
   dataset: Dataset;
+  datasourceName: string;
 }): Promise<AssetReference> {
   const mappings = generateMappings(fields);
   const templateName = generateTemplateName(dataset);
-  const template = getTemplate(templateName + '-*', mappings);
+  let pipelineName;
+  if (dataset.ingest_pipeline) {
+    pipelineName = getPipelineNameForInstallation(
+      dataset.ingest_pipeline,
```

Comment:
Can't we get rid of the first and third param as we already pass along the dataset?

Reply:
The name of the pipeline is not always `dataset.ingest_pipeline` because the pipeline we request the name for might not be the default pipeline.

The change that `dataset.package` exists came in through a rebase, this can be cleaned up.

Reply:
Agree, the name is not always `dataset.ingest_pipeline`. But then I don't fully understand in which case it will be different as we pass here `dataset.ingest_pipeline` for the name.

Reply:
>  the pipeline we request the name for might not be the default pipeline.

I mean, it might not be the entry pipeline in case there are more pipelines and they refer to each other.

Reply:
We just clarified it in a zoom call. The part I missed is that `getPipelineNameForInstallation` is also used in other places.

Reply:
(insert zoom discussion here, for posterity:)

The function `getPipelineNameForInstallation()` is called from two places.
* from the template installation, where the pipeline name is always `dataset.ingest_pipeline`
* from the pipeline installation, where the pipeline name can be different when a dataset contains more than one pipeline. In that case, the name is taken from the assed path.